### ISSUE TITLE
Firefly III DB crash after 4.7.1.4 update

### ISSUE THREAD

**Issue by 8bitgentleman** [firefly-iii/firefly-iii#1241]
I am running Firefly III version 4.7.1.4

#### Description of my issue:
I am running firefly via docker. The install and update instructions worked just fine with previous updates. Today after updating to 4.7.1.4 my firefly db immediately crashed and returns [this error](https://gist.github.com/8bitgentleman/c0c950495eb57d9ed89594601d883064). I attempted to reinstall firefly but the same crash and error occur with all 3 install options

#### Steps to reproduce 
Update/install 4.7.1.4 with docker

**Comment by JC5** [comment]
I have no clue what could cause this. I'll will try to reproduce it.

**Comment by JC5** [comment]
No luck so far. From which version did you upgrade, and what method did you use exactly?

**Comment by 8bitgentleman** [comment]
Originally I was updating from 4.7.0 with docker compose when I got the error. I have also tried a fresh install of 4.7.1.4 using all 3 docker methods and got the same error.

**Comment by JC5** [comment]
I know that mysql:8 and PHP don't play nice any more, because of an upgrade in MySQL. I've changed this between 4.7.0 and 4.7.1.4.

This has been changed after 4.7.0, because it turned out that PHP was unable to contact MySQL. I had not expected format changes between the files. 

I would suggest extracting your DB from the old container and re-upping it into a new one. 4.1.7.4 should use the correct MySQL version.

Make sure you make a backup before you continue. Then, have you tried to run `docker-compose pull` already?

**Comment by 8bitgentleman** [comment]
Trying `docker-compose pull` says that both mysql and firefly iii are up to date from their respective :latest branches. I have saved my DB from the old container but after creating brand new installs, using the docker-compose method, the db container just fails immediately after creation. Should I try to target another version of mysql when I build the containers?

**Comment by JC5** [comment]
I'm using :latest as well.

```
Mac-mini:issue-1241 james$ docker-compose pull
Pulling firefly_iii_db (mysql:latest)...
latest: Pulling from library/mysql
Digest: sha256:227d5c3f54ee3a70c075b1c3013e72781564000d34fc8c7ec5ec353c5b7ef7fa
Status: Image is up to date for mysql:latest
Pulling firefly_iii_app (jc5x/firefly-iii:latest)...
latest: Pulling from jc5x/firefly-iii
Digest: sha256:6ab64391700b93eff591fc062c4412aba1cde7a12b93601008c1096a0d4412e8
Status: Image is up to date for jc5x/firefly-iii:latest
Mac-mini:issue-1241 james$
```

To make sure I was running the latest image (see bottom right):

```
docker-compose down
docker-compose -f docker-compose.yml up --build --no-deps -d
```

Not that of a Docker guru though, not sure if it was necessary.

**Comment by 8bitgentleman** [comment]
Ok I figured it out and I'm relatively certain I know what was going on. I got rid of both the app and db containers than ran `docker system prune --force` to get rid of any old images that were laying around. That ran for a bit and deleted what, I assume, was either an old firefly iii or mysql image.

I then ran `docker-compose -f docker-compose.yml up --build --no-deps -d` like you suggested and things went smoothly from there. 

Thanks for your help!
